mongo/integration/client_side_encryption_prose_test.go

```diff
@@ -896,6 +897,119 @@ func TestClientSideEncryptionProse(t *testing.T) {
 			}
 		})
 	})
+
+	testcases := []struct {
+		description                            string
+		maxPoolSize                            uint64
+		bypassAutoEncryptionSet                bool
+		keyVaultClientSet                      bool
+		clientEncryptedTopologyOpeningExpected int
+		clientEncryptedCommandStartedExpected  []startedEvent
+		clientKeyVaultCommandStartedExpected   []startedEvent
+	}{
+		{
+			"deadlock case 1", 1, false, false, 2,
+			[]startedEvent{{"listCollections", "db"}, {"find", "keyvault"}, {"insert", "db"}, {"find", "db"}},
+			nil,
+		},
+		{
+			"deadlock case 2", 1, false, true, 2,
+			[]startedEvent{{"listCollections", "db"}, {"insert", "db"}, {"find", "db"}},
+			[]startedEvent{{"find", "keyvault"}},
+		},
+		{
+			"deadlock case 3", 1, true, false, 2,
+			[]startedEvent{{"find", "db"}, {"find", "keyvault"}},
+			nil,
+		},
+		{
+			"deadlock case 4", 1, true, true, 1,
+			[]startedEvent{{"find", "db"}},
+			[]startedEvent{{"find", "keyvault"}},
+		},
+		{
+			"deadlock case 5", 0, false, false, 1,
+			[]startedEvent{{"listCollections", "db"}, {"listCollections", "keyvault"}, {"find", "keyvault"}, {"insert", "db"}, {"find", "db"}},
+			nil,
+		},
+		{
+			"deadlock case 6", 0, false, true, 1,
+			[]startedEvent{{"listCollections", "db"}, {"insert", "db"}, {"find", "db"}},
+			[]startedEvent{{"find", "keyvault"}},
+		},
+		{
+			"deadlock case 7", 0, true, false, 1,
+			[]startedEvent{{"find", "db"}, {"find", "keyvault"}},
+			nil,
+		},
+		{
+			"deadlock case 8", 0, true, true, 1,
+			[]startedEvent{{"find", "db"}},
+			[]startedEvent{{"find", "keyvault"}},
+		},
```

Comment:
The spec mentions a possible deadlock with `maxPoolSize=2`; should you add a case for that?

Reply:
Sounds good; thanks for the context!